src/main/java/org/jenkinsci/plugins/pipeline/github/CommitStatusGroovyObject.java

```diff
@@ -18,55 +19,48 @@
         this.commitStatus = commitStatus;
     }
 
-    @Override
-    public Object getProperty(final String property) {
-        if (property == null) {
-            throw new MissingPropertyException("null", this.getClass());
-        }
+    @Whitelisted
+    public String getCreator() {
+        return commitStatus.getCreator().getLogin();
+    }
+
+    @Whitelisted
+    public Date getCreatedAt() {
+        return commitStatus.getCreatedAt();
+    }
 
-        switch (property) {
-            case "id":
-                return commitStatus.getId();
-            case "url":
-                return commitStatus.getUrl();
-            case "status":
-                return commitStatus.getState();
-            case "context":
-                return commitStatus.getContext();
-            case "description":
-                return commitStatus.getDescription();
-            case "target_url":
-                return commitStatus.getTargetUrl();
-            case "created_at":
-                return commitStatus.getCreatedAt();
-            case "updated_at":
-                return commitStatus.getUpdatedAt();
-            case "creator":
-                return commitStatus.getCreator().getLogin();
-            default:
-                throw new MissingPropertyException(property, this.getClass());
-        }
+    @Whitelisted
+    public Date getUpdatedAt() {
+        return commitStatus.getUpdatedAt();
     }
 
-    @Override
-    public void setProperty(final String property, final Object newValue) {
-        if (property == null) {
-            throw new MissingPropertyException("null", getClass());
-        }
+    @Whitelisted
+    public long getId() {
+        return commitStatus.getId();
+    }
+
+    @Whitelisted
+    public String getContext() {
+        return commitStatus.getContext();
+    }
+
+    @Whitelisted
+    public String getDescription() {
+        return commitStatus.getDescription();
+    }
+
+    @Whitelisted
+    public String getState() {
```

Comment:
This was previously available as `status`. It should either be changed to `getStatus` or documentation should be updated.